src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,83 @@
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private final Map<Class<?>, Object> instances = new HashMap<>();
+
+    private final Map<Class<?>, Class<?>> interfaceImplementations = new HashMap<>() {{
+            put(ProductService.class, ProductServiceImpl.class);
+            put(FileReaderService.class, FileReaderServiceImpl.class);
+            put(ProductParser.class, ProductParserImpl.class);
+        }};
 
     public static Injector getInjector() {
         return injector;
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        if (instances.containsKey(interfaceClazz)) {
+            return instances.get(interfaceClazz);
+        }
+
+        Class<?> clazz = findImplementation(interfaceClazz);
+        if (!clazz.isAnnotationPresent(Component.class)) {
+            throw new RuntimeException("Class " + clazz.getName()
+                    + " is not marked with @Component");
+        }
+
+        Object clazzImplementationInstance = createNewInstance(clazz);
+        injectDependencies(clazzImplementationInstance);
+        instances.put(interfaceClazz, clazzImplementationInstance);
+        return clazzImplementationInstance;
+    }
+
+    private Object createNewInstance(Class<?> clazz) {
+        try {
+            Constructor<?> constructor = clazz.getConstructor();
+            return constructor.newInstance();
+        } catch (NoSuchMethodException | InvocationTargetException
+                 | InstantiationException | IllegalAccessException e) {
+            throw new RuntimeException("Can't create a new instance of " + clazz.getName(), e);
+        }
+    }
+
+    private void injectDependencies(Object instance) {
+        Field[] declaredFields = instance.getClass().getDeclaredFields();
+        for (Field field : declaredFields) {
+            if (field.isAnnotationPresent(Inject.class)) {
+                Object fieldInstance = getInstance(field.getType());
+                try {
+                    field.setAccessible(true);
+                    field.set(instance, fieldInstance);
+                } catch (IllegalAccessException e) {
+                    throw new RuntimeException("Can't initialize field value. Class: "
+                            + instance.getClass().getName() + ", Field: " + field.getName(), e);
+                }
+            }
+        }
+    }
+
+    private Class<?> findImplementation(Class<?> interfaceClazz) {
+        if (interfaceClazz.isInterface()) {
+            Class<?> implClass = interfaceImplementations.get(interfaceClazz);
+            if (implClass == null) {
+                throw new RuntimeException("No implementation found for: "
+                        + interfaceClazz.getName());
+            }
```

Comment:
Make sure that the `interfaceImplementations` map is fully populated with all necessary interface-implementation pairs. If an interface is not mapped, it will result in a `RuntimeException` when trying to find its implementation.